### PR TITLE
GUI: move game settings out of Engine tab

### PR DESCRIPTION
**Pull request by Die4Ever**
I think these options are a bit too hidden for most users to find. Moving them to the Game tab ensures everyone will see the options immediately upon adding a game. This is great for an option like "Easier AI" for The 11th Hour, where not knowing about this option could cause a lot of needless frustration for the player.

Currently there's extra spacing that I don't know how to get rid of, I'm sure it's something very simple. But at least it works!

image

### CODE REVIEW

File: gui/editgamedialog.cpp [hunk 1]
```diff
@@ -184,25 +184,23 @@ EditGameDialog::EditGameDialog(const Common::String &domain)
...
 	if (metaEnginePlugin) {
-		int tabId = tab->addTab(_("Engine"), "GameOptions_Engine");
+		tab->setActiveTab(0);
 
 		const MetaEngineDetection &metaEngineDetection = metaEnginePlugin->get<MetaEngineDetection>();
 		metaEngineDetection.registerDefaultSettings(_domain);
 		if (enginePlugin) {
 			enginePlugin->get<MetaEngine>().registerDefaultSettings(_domain);
-			_engineOptions = enginePlugin->get<MetaEngine>().buildEngineOptionsWidgetDynamic(tab, "GameOptions_Engine.Container", _domain);
+			_engineOptions = enginePlugin->get<MetaEngine>().buildEngineOptionsWidgetDynamic(tab, "GameOptions_Game.Container", _domain);
```
Comment (Die4Ever) [inline gui/editgamedialog.cpp:197]: we could even keep buildEngineOptionsWidgetDynamic using the separate Engine tab if we want, or make an additional function for that

**Comment by sev-** [comment]
This will not work on a low resolution like 320x200. You could, however, potentially, split this code depending on the resolution.

**Comment by Die4Ever** [comment]
> This will not work on a low resolution like 320x200. You could, however, potentially, split this code depending on the resolution.

now with scrollbar!

image

and I fixed the spacing issue

image

File: gui/themes/common/lowres_layout.stx [hunk 2]
```diff
@@ -1266,6 +1266,12 @@
...
 
 	<dialog name = 'GameOptions_Game' overlays = 'Dialog.GameOptions.TabWidget' shading = 'dim'>
+		<layout type = 'vertical' padding = '0, 0, 0, 0' spacing = '0'>
+			<widget name = 'Container'/>
+		</layout>
+	</dialog>
+	
+	<dialog name = 'GameOptions_Game_Container' overlays ='Dialog.GameOptions_Game.Container' inset='0'>
 		<layout type = 'vertical' padding = '8, 8, 8, 8'>
 			<layout type = 'horizontal' padding = '0, 0, 0, 0' spacing = '6' align = 'center'>
```
Comment (Die4Ever) [inline gui/themes/common/lowres_layout.stx:1276]: now that we have a scrollbar, we could remove these horizontal layouts from lowres if we wanted to

**Comment by sev-** [comment]
Looks good. However, 2 things are missing:
 - Bump theme version (THEMERC file and `gui/ThemeEngine.h` file)
 - Update classic theme

**Comment by Die4Ever** [comment]
this is probably superseded by #3594 although that one is a different approach and more complex

**Comment by bluegr** [comment]
> this is probably superseded by #3594 although that one is a different approach and more complex

You mean one part of this pull request, right? Cause one pull request adds tab scrolling and the other moves options to another tab

**Comment by Die4Ever** [comment]
> > this is probably superseded by #3594 although that one is a different approach and more complex
> 
> You mean one part of this pull request, right? Cause one pull request adds tab scrolling and the other moves options to another tab

No, the other pull request does that too, it just doesn't require as large of a change in the themes files to accomplish it

**Comment by bluegr** [comment]
Closing this one in favor of #3594, which provides more flexibility

**Pull request closed without merging**